File: project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/model/inheritance/AdministratorBase.java
package org.babyfish.jimmer.sql.model.inheritance;

import org.babyfish.jimmer.sql.*;
import org.jspecify.annotations.Nullable;

import java.util.List;

@MappedSuperclass
public interface AdministratorBase extends NamedEntity {

    @ManyToMany
    @JoinTable(
            name = "ADMINISTRATOR_ROLE_MAPPING",
            joinColumnName = "ADMINISTRATOR_ID",
            inverseJoinColumnName = "ROLE_ID",
            deletedWhenEndpointIsLogicallyDeleted = true
    )
    List<Role> getRoles();

    @OneToOne(mappedBy = "administrator")
    @Nullable
    AdministratorMetadata getMetadata();
}

File: project/jimmer-sql-kotlin/src/test/kotlin/org/babyfish/jimmer/sql/kt/model/inheritance/AdministratorBase.kt
package org.babyfish.jimmer.sql.kt.model.inheritance

import org.babyfish.jimmer.sql.*

@MappedSuperclass
interface AdministratorBase : NamedEntity {

    @ManyToMany
    @JoinTable(
        name = "ADMINISTRATOR_ROLE_MAPPING",
        joinColumnName = "ADMINISTRATOR_ID",
        inverseJoinColumnName = "ROLE_ID"
    )
    val roles: List<Role>

    @OneToOne(mappedBy = "administrator")
    val metadata: AdministratorMetadata?
}